selftests/utils.py

```diff
@@ -146,7 +148,7 @@ def skipUnlessPathExists(path):
     )
 
 
-class TestCaseTmpDir(unittest.TestCase):
+class TestCaseTmpDir(Test):
```

Comment:
The reason for `TestCaseTmpDir` was to use the "lighter" `unittest.TestCase` with a temporary dir.  If we change that to use `avocado.Test`, we should really be using `self.workdir` instead of adding another temporary directory.

Instead of changing all of Avocado's selftests into `avocado.Test` based tests, I'd rather segregate the `unittest.TestCase` based (including `TestCaseTmpDir` ones) and the `avocado.Test` based tests into separate files.  I believe there's value in making sure Avocado itself uses a variation of test types, as this is something we advertise and support.

Reply:
> Instead of changing all of Avocado's selftests into `avocado.Test` based tests, I'd rather segregate the `unittest.TestCase` based (including `TestCaseTmpDir` ones) and the `avocado.Test` based tests into separate files. I believe there's value in making sure Avocado itself uses a variation of test types, as this is something we advertise and support.

I see your point about advertising support of multiple types of tests, but when I tried to segregate the tests based on the type, I found out that it makes the selftests more complex and creating new tests more complicated. 

Right now, we have one unit test file and one functional test file for each feature/module (this is not precise but more general description) so we have two test files fore each module. When we segregate the tests, we will have four test files for each module. With such change, you will have to check twice as many files always when you want to make some changes to tests.

If you think that changing the tests to `avocado.Test` is a bad idea. Then I would propose to change the resolver behavior instead of changing the selftests.

Reply:
Well, my main point is about *not* turning `TestCaseTmpDir` into a child of `avocado.Test`.  This would ensure we have a good number of `unittest.TestCase` based tests in the selftests.

But, looking at a single file at a time, it's fair game to make all tests within it based on `avocado.Test` *if* there's one based on it already (and if it makes sense for it to be based on `avocado.Test`).